tests/Integration/Buffer/Tests/TestIsSpeedTool.php

```diff
@@ -5,13 +5,21 @@
 use WP_Rocket\Buffer\Tests;
 use WP_Rocket\Buffer\Config;
 
+/**
+ * @group Buffer
+ */
 class TestIsSpeedTool extends TestCase {
      /**
      * @covers ::is_speed_tool
      * @author Remy Perona
      */
     public function testShouldReturnTrueWhenLighthouse() {
-        $config = new Config(
+    	// Grab the current Config::$config_dir_path value. We'll restore it when we're done.
+	    $config_dir_path = $this->get_reflective_property( 'config_dir_path', 'WP_Rocket\Buffer\Config' );
+	    // Set the Config::$config_dir_path value to `null`.
+	    $this->set_reflective_property( null, 'config_dir_path', 'WP_Rocket\Buffer\Config' );
```

Comment:
When the plugin boots up, the `Config::$config_dir_path` is set statically. When we go to run this test, it's not using the configuration below, but rather what was set on boot up. Okay, that's normal for production. 

How do we handle it here? With reflection.

The trait I added gives us the ability to access private and protected properties. Aha. That's really handy in cases like this one.

Here:

1. Grab the value that's been set in the static property.
2. Set the property to `null`, which is what we need for this test.
3. Run the test.
4. Restore the original value.

Reply:
Does it mean that the test was looking fine before, but in fact it was not really reliable?

Reply:
Yes.

It was working until:

- the plugin loaded as it does in production
- we built another integration test that preloaded the config. If the new test ran first, then the speed tool test would fail

Using the above strategy, we eliminated any race/timing issues and ensured the state is what we need specifically for the context of this one test and edge case.